src/vehicle.cpp

```diff
@@ -4693,7 +4773,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+            if( chainsaw_on && part_info( ret.part ).has_flag( "CHAINSAW" ) &&
+                parts[part].hp > 0 ) {
+                dam += 66;
```

Comment:
I've tried to implement Coolthulhu's proposition about ```on_move``` & ```idle``` functions and I realized I have no ideas how to detect moments of interaction between a chainsaw and a critter so I'll try to switch to Kevin proposition.

BTW, I found ```part.enabled``` variable so I want to track status of each chainsaw using it.